.github/workflows/build-master.yaml

```diff
@@ -0,0 +1,61 @@
+name: build-master
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '.mvn/**'
+      - 'mvnw'
+      - 'mvnw.cmd'
+      - 'README.md'
+      - '.gitignore'
+
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '.mvn/**'
+      - 'mvnw'
+      - 'mvnw.cmd'
+      - 'README.md'
+      - '.gitignore'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'corretto'
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Start Database
+        run: ./setup-database.bash
+
+      - name: Build project
+        run: mvn clean install
+
+      - name: Show test result
+        run: mvn surefire-report:report-only
+             cat target/site/surefire-report.html
+
+      - name: Publish to GitHub Packages
```

Comment:
Why we need it?

Reply:
Im not sure if we need to publish image, we for now not doing auto deploys, we just need a builds, if im mistake improve me :D

Reply:
to build docker images in https://github.com/karunaproject/devops in dockerfile